flare/kernels/kernels.py

```diff
@@ -277,6 +277,33 @@ def q_value_mc(distances, r_cut, ref_species, species, cutoff_func, q_func=coord
     return q
 
 
+@njit
+def q_value_mc(distances, r_cut, ref_species, species, cutoff_func, q_func=coordination_number):
+    """Compute value of many-body many components descriptor based
+    on distances of atoms in the local many-body environment.
+
+    Args:
+        distances (np.ndarray): distances between atoms i and j
+        r_cut (float): cutoff hyperparameter
+        ref_species (int): species to consider to compute the contribution
+        species (np.ndarray): atomic species of neighbours
+        cutoff_func (callable): cutoff function
+        q_func (callable): many-body pairwise descrptor function
+
+    Return:
+        float: the value of the many-body descriptor
+    """
+
+    q = 0
+
+    for i in range(len(distances)):
+        if species[i] == ref_species:
+            q_, _ = q_func(distances[i], 0, r_cut, cutoff_func)
+            q += q_
```

Comment:
Never saw the convention of temporary variables being underlined e.g. `q_`, but I like it! :)